frontend/src/libs/components/button/button.tsx

```diff
@@ -1,28 +1,42 @@
+import { type IconName } from '#libs/types/icon-name.type.js';
+
+import { Icon } from '../components.js';
 import styles from './styles.module.scss';
 
 type Properties = {
-  label: string;
+  label?: string;
   type?: 'button' | 'submit';
-  style?: 'primary' | 'secondary';
+  iconName?: IconName;
+  style?: 'primary' | 'secondary' | 'rounded' | 'rounded-transparent';
   isLoading?: boolean;
   isDisabled?: boolean;
+  accessabilityMessage?: string;
+  onClick?: () => void;
 };
 
 const Button: React.FC<Properties> = ({
   type = 'button',
   label,
+  iconName,
   style = 'primary',
   isLoading = false,
   isDisabled = false,
+  accessabilityMessage = '',
```

Comment:
let's pass flag isLabelVisuallyHidden instead of accessabilityMessage and make label required back
```suggestion
  isLabelVisuallyHidden = false,
```